#! /bin/sh /usr/share/dpatch/dpatch-run
## 03_fix_cow_tmpfs_mount.dpatch by  <mika@grml.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix cow tmpfs mount option handling

@DPATCH@

From afccb65c91dcf8a04f2c8a4969d3e7799087bfc6 Mon Sep 17 00:00:00 2001
From: Michael Prokop <mika@grml.org>
Date: Thu, 17 Jul 2008 11:28:30 +0200
Subject: [PATCH 1/1] Make sure mounting /cow-tmpfs provides mount options


Signed-off-by: Michael Prokop <mika@grml.org>

diff --git a/scripts/live b/scripts/live
index 15710c2..22d2210 100755
--- a/scripts/live
+++ b/scripts/live
@@ -1118,6 +1118,8 @@ setup_unionfs ()
 		nfsmount ${nfs_cow_opts} ${cowdevice} /cow || \
 			panic "Can not mount ${cowdevice} (n: ${cow_fstype}) on /cow"
 	else
+		# make sure cow_mountopt is set:
+		[ -n "$cow_mountopt" ] || cow_mountopt="rw,noatime,mode=755"
 		mount -t ${cow_fstype} -o ${cow_mountopt} ${cowdevice} /cow || \
 			panic "Can not mount ${cowdevice} (o: ${cow_fstype}) on /cow"
 	fi
-- 
1.5.5.4

